### ISSUE TITLE
Jakarta - Avoid test classpath pollution

### ISSUE THREAD

**Issue by gsmet** [quarkusio/quarkus#27593]
This is a preexisting issue: the test classpath is polluted by the Bootstrap dependencies, especially those coming from the Maven stuff.

It is even more annoying with the Jakarta migration as we will have a javax.inject dependency coming with Maven (we cannot get rid of it, unfortunately), which means that we will have several `@Inject` annotations around.

@aloubyansky is going to experiment with a possible solution to this issue.

**Comment by geoand** [comment]
Has this been addressed?

**Comment by manovotn** [comment]
I think this is still the case but I recall we detect wrong usage of those annotations now, see https://github.com/quarkusio/quarkus/blob/main/extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/WrongAnnotationUsageProcessor.java#L38-L50

**Comment by aloubyansky** [comment]
No, it's not. I have a branch where I made it work for QuarkusTest case but it's too complicated. I thought I should try a different approach.

**Comment by aloubyansky** [comment]
Yes, I agree @gsmet
